# Exploratory Data Analysis

This notebook explores the initial public market dataset before feature engineering or modeling. The first section independently loads the saved Parquet file and checks its dimensions, coverage, missing values, and duplicate observations.

In [1]:
from pathlib import Path

import pandas as pd

current_directory = Path.cwd()

if current_directory.name == "notebooks":
    project_root = current_directory.parent
else:
    project_root = current_directory

data_path = (
    project_root
    / "data"
    / "raw"
    / "daily_prices_2016_2025.parquet"
)

data_path

WindowsPath('c:/Users/User/OneDrive/Documents/Quant Projects/jane-street-inspired-market-prediction/data/raw/daily_prices_2016_2025.parquet')

In [2]:
prices = pd.read_parquet(data_path)

print("Dataset loaded successfully.")

Dataset loaded successfully.


In [3]:
row_count, column_count = prices.shape
unique_dates = prices["date"].nunique()
unique_symbols = prices["symbol"].nunique()

print(f"Rows: {row_count:,}")
print(f"Columns: {column_count}")
print(f"First date: {prices['date'].min().date()}")
print(f"Last date: {prices['date'].max().date()}")
print(f"Unique trading dates: {unique_dates:,}")
print(f"Unique symbols: {unique_symbols}")

Rows: 77,934
Columns: 7
First date: 2016-01-04
Last date: 2025-12-31
Unique trading dates: 2,514
Unique symbols: 31


In [4]:
missing_summary = pd.DataFrame(
    {
        "missing_count": prices.isna().sum(),
        "missing_percent": prices.isna().mean() * 100,
    }
)

missing_summary

,missing_count,missing_percent
date,0,0.0
symbol,0,0.0
open,0,0.0
high,0,0.0
low,0,0.0
close,0,0.0
volume,0,0.0


In [5]:
duplicate_count = prices.duplicated(
    subset=["date", "symbol"]
).sum()

print(f"Duplicate date-symbol rows: {duplicate_count:,}")

Duplicate date-symbol rows: 0


In [6]:
expected_row_count = unique_dates * unique_symbols
panel_is_complete = row_count == expected_row_count

print(f"Expected rows in a complete panel: {expected_row_count:,}")
print(f"Actual rows: {row_count:,}")
print(f"Complete date-symbol panel: {panel_is_complete}")

Expected rows in a complete panel: 77,934
Actual rows: 77,934
Complete date-symbol panel: True


## Initial Quality Audit

The dataset loaded successfully and contains 77,934 observations covering 2,514 trading dates and 31 symbols. No columns contain missing values, no duplicate date-symbol keys were found, and the row count matches a complete panel containing one observation for every symbol on every trading date.